src/libs/auth.ts

```diff
@@ -97,71 +127,193 @@ const revokeTokens = async () => {
+      totalTokenLoadAttemptsThisSession: authStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
     },
   }));
-  Ajax().Metrics.captureEvent(Events.userLogin, {
-    authProvider: user.profile.idp,
-    sessionStartTime: Utils.makeCompleteDate(sessionStartTime),
-    authTokenCreatedAt: Utils.formatTimestampInSeconds(authTokenCreatedAt),
-    authTokenExpiresAt: Utils.formatTimestampInSeconds(authTokenExpiresAt),
-    jwtExpiresAt: Utils.formatTimestampInSeconds(jwtExpiresAt),
+
+  const loadedAuthTokenState: AuthTokenState = await tryLoadAuthToken(includeBillingScope, popUp);
+
+  if (loadedAuthTokenState.status === 'success') {
+    const user: User = (loadedAuthTokenState as AuthTokenSuccessState).user;
+    const userJWT: string = user.id_token!;
+    const decodedJWT: JwtPayload = jwtDecode<JwtPayload>(userJWT);
+    const authTokenCreatedAt: number = (decodedJWT as any).auth_time; // time in seconds when authorization token was created
+    const authTokenExpiresAt: number = user.expires_at!; // time in seconds when authorization token expires, as given by the oidc client
+    const authTokenId: string = uuid();
+    const jwtExpiresAt: number = (decodedJWT as any).exp; // time in seconds when the JWT expires, after which the JWT should not be read from
+    const refreshToken: string | undefined = user.refresh_token;
+    // for future might want to take refresh token and hash it to compare to a saved hashed token to see if they are the same
+    const isNewRefreshToken = !!refreshToken && refreshToken !== oldRefreshTokenMetadata.token;
+
+    // for metrics, updates authToken metadata and refresh token metadata
+    authStore.update((state) => ({
+      ...state,
+      authTokenMetadata: {
+        ...authStore.get().authTokenMetadata,
+        token: user.access_token,
+        createdAt: authTokenCreatedAt,
+        expiresAt: authTokenExpiresAt,
+        id: authTokenId,
+        totalTokensUsedThisSession:
+          oldAuthTokenMetadata.token !== undefined &&
+          oldAuthTokenMetadata.token === authStore.get().authTokenMetadata.token
+            ? oldAuthTokenMetadata.totalTokensUsedThisSession
+            : oldAuthTokenMetadata.totalTokensUsedThisSession + 1,
+      },
+      refreshTokenMetadata: isNewRefreshToken
+        ? {
+            ...authStore.get().refreshTokenMetadata,
+            id: uuid(),
+            token: refreshToken,
+            createdAt: authTokenCreatedAt,
+            // Auth token expiration is set to 24 hours in B2C configuration.
+            expiresAt: authTokenCreatedAt + 86400, // 24 hours in seconds
+            totalTokensUsedThisSession: authStore.get().authTokenMetadata.totalTokensUsedThisSession + 1,
+            totalTokenLoadAttemptsThisSession: authStore.get().authTokenMetadata.totalTokenLoadAttemptsThisSession + 1,
+          }
+        : { ...authStore.get().refreshTokenMetadata },
+    }));
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
```

Comment:
To @gpolumbo-broad's [earlier comment](https://github.com/DataBiosphere/terra-ui/pull/4171#discussion_r1328806147), it looks like sending the metric requests could easily be split out into a separate function. It looks like most of the information they use is stored in `authStore`.